lib/ansible/modules/cloud/amazon/aws_guardduty_detector.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Aaron Smith <ajsmith10381@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: aws_guardduty_detector
+short_description: Enable/Disable the AWS GuardDuty detector.
+description:
+    - Enable/Disable the GuardDuty detector for a given AWS account.
+author: "Aaron Smith (@slapula)"
+version_added: "2.7"
+requirements: [ 'botocore', 'boto3' ]
+options:
+  state:
+    description:
+     - Whether the detector should be exist or not.
+    required: true
+    choices: ['present', 'absent']
+  enabled:
+    description:
+     - Whether the detector should be enabled or disabled.
+    type: bool
+    default: true
+extends_documentation_fragment:
+    - ec2
+    - aws
+notes:
+  - Currently, GuardDuty supports only one detector resource per AWS account per region.
+'''
+
+
+EXAMPLES = r'''
+- name: create GuardDuty detector
+  guardduty_detector:
+    state: present
+    enabled: true
+
+- name: disable GuardDuty detector
+  guardduty_detector:
+    state: present
+    enabled: false
+
+- name: delete GuardDuty detector
+  guardduty_detector:
+    state: absent
+'''
+
+
+RETURN = r'''
+detector_id:
+    description: The ID of the GuardDuty detector you just created or updated.
+    returned: always
+    type: string
+'''
+
+import os
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def detector_exists(client, module, result):
+    try:
+        response = client.list_detectors()
+        return {'exists': True, 'detector_id': response['DetectorIds'][0]}
+    except (ClientError, IndexError):
```

Comment:
When I call list_detectors and there are no detectors it returns an empty list `{..., 'DetectorIds': []}`, so you can keep the `except IndexError:` and return False there, but ClientError and BotoCoreError exceptions should be handled with fail_json_aws, rather than masking or not handling those.